Added Basic Class-Level SAS Capability to checkSAS, added Vector4f to SAS

DaemonUmbra opened the pull request:
Feel free to cherry pick just the checkSAS change

shartte committed:
Sets the empty/fill sounds for vanilla fluids.
Use Fluid's fillSound rather than emptySound when filling buckets.

cpw committed:
Revert "Shut down all other mod handlers if the loading cycle errors. This prevents other mods from throwing errors and being blamed for initial cause. This is a temporary hack until cpw re-writes the mod event dispatcher."

This reverts commit 7592bbe8

Signed-off-by: cpw <cpw+github@weeksfamily.ca>

cpw committed:
Revert "Properly shutdown FMLModContainer's event bus when an error in a lifecycle event is detected."

This reverts commit 30bad1e2

Signed-off-by: cpw <cpw+github@weeksfamily.ca>

DaemonUmbra committed:
Added Class-level SAS capability to checkSAS

DaemonUmbra committed:
Added Class-level SAS capability to checkSAS

DaemonUmbra committed:
Added Vector4f class to SAS

All the other Vectors are common, might as well finish the job

DaemonUmbra committed:
Added Vector4f class to SAS

All the other Vectors are common, might as well finish the job

LexManos commented on build.gradle:
```diff
@@ -882,11 +882,18 @@ project(':forge') {
+
+                    //Class SAS
+                    if(desc.isEmpty()) {
+                        lines.add(cls + (comment == null ? '' : ' ' + comment))
```
Space after if, aand use the json to make sure the class exists. It's the point of validating...
Also, should this propagate to children?
Should this propagate all overridden methods? Such as a Sided interface being stripped, thus stripping the needing the implementations of that interface to be revived. I'm thinking IChestLid here.

DaemonUmbra replied:
I was wondering if it should propagate to children.
At the time I just wanted to be able to get Vector4f common before RB, should have marked this as a draft

DaemonUmbra replied:
Would be nice to have it propagate but I'm not sure checkSAS has enough info to be sure that the child of one SASed interface doesn't implement other sideonly interfaces that prevent it from being a common class

LexManos replied:
It has enough information to make sure you can propagate it to the specific methods inside the class you're unsiding.
It doesn't need to unside the whole class of the client. Just the methods.

DaemonUmbra committed:
Made class-level SAS lines propagate to their methods and methods that override

DaemonUmbra committed:
Made class-level SAS lines propagate to their methods and methods that override

gigaherz committed:
Fix ModelRegistryEvent firing every time resources reload instead of just once.

LexManos committed:
Initial 1.16.2 Update

Cyborgmas committed:
Fix server connection, misapplied patch (#7245)

Yunus1903 committed:
Updated MDK and README for 1.16.2 (#7243)

spnda committed:
Print mod file name on InvalidModFileException (#7241)

LexManos committed:
Fix creating nether portals.

LexManos committed:
New method for creating modded tag references, fix connecting to vanilla servers.

Cyborgmas committed:
Remove forge optional tags in favor of Vanilla's new system. (#7246)

Yunus1903 committed:
Removed HarvestDropsEvent (#7193)

malte0811 committed:
Improve startup time by caching the manifest data for mod jars (#7256)

ChampionAsh5357 committed:
Readding DifficultyChangeEvent hooks (#7240)

LexManos committed:
Cleanup Forge's build script, fix some unneeded patches.
Add checkExcscheckAll tasks.

diesieben07 committed:
Fix config iteration order (#7230)

Cyborgmas committed:
Fix small logging issues with loading pack.png for mods (#7265)

diesieben07 committed:
Improve performance and cleanup code for DelegatingResourcePack (#7228)

mcenderdragon committed:
Fix ItemStack#isDamageable is not calling Item#isDamagable (#7221)

phit committed:
Log info about server list warning (#7209)

Make it easier to figure out what mods are missing or have to be marked as client/server side only.

sciwhiz12 committed:
Fix modproperties property in mods.toml causing exception (#7192)

Cyborgmas committed:
Cleanup and expand Forge tags (#7004)

Edivad99 committed:
[1.16] ForgeEventFactory#canCreateFluidSource reintroduced (#7181)

maxanier committed:
Replace EntityHeight event with EntitySize event (#6858)

sciwhiz12 committed:
Add user-friendly exceptions when config loading fails (#7214)

gigaherz committed:
Clarify in the LazyOptional which methods carry the lazyness over to the returned value and which don't. (#6750)

For consistency, this meant making a few changes:
- Renamed the existing lazy mapping method to lazyMap, to indicate that it doesn't run the mapping immediately.
- Added a new implementation of map(), which returns Optional<T>, and resolves the value in the process.
- Changed filter() to return Optional, since there's no way to filter lazily.
- Added a new method resolve(), which helps convert the custom LazyOptional class into a standard Optional, for use
with library methods that expect Optional<T>.

* Update License headers.

LexManos committed:
Wrap vanilla's 'default' biome registry in ForgeRegistry.
This should allow registering mod dimensions like in 1.16.1, while the data driven system is fleshed out.

ichttt committed:
Add particle culling (#6981)

cpw committed:
Redo event dispatch, removes a bunch of nonsense, and tries to integrate with the vanilla CF system where possible

Signed-off-by: cpw <cpw+github@weeksfamily.ca>

cpw committed:
Merge branch '1.16.x' into 1.16.1

# Conflicts:
#	src/main/java/net/minecraftforge/fml/ModList.java
#	src/main/java/net/minecraftforge/fml/client/ClientModLoader.java

cpw committed:
Fix up all the things. Removed SidedProvider because it served no real purpose anymore.

Signed-off-by: cpw <cpw+github@weeksfamily.ca>

cpw committed:
Merge branch '1.16.1' into 1.16.x

# Conflicts:
#	patches/minecraft/net/minecraft/client/Minecraft.java.patch
#	patches/minecraft/net/minecraft/item/crafting/Ingredient.java.patch
#	src/main/java/net/minecraftforge/fml/client/ClientModLoader.java
#	src/main/java/net/minecraftforge/registries/GameData.java

cpw committed:
Fix up slight registry change and other 1.16.2 stuffs.

Signed-off-by: cpw <cpw+github@weeksfamily.ca>

cpw committed:
Fix missed debugging code. Read the config from the config.

Signed-off-by: cpw <cpw+github@weeksfamily.ca>

cpw committed:
Move the "modloading" thread activity onto _our_ worker pool. It turns out that the vanilla worker pool can deadlock during stitching if it's insufficiently large, if modloader "waitForTransition" is also a worker member of that pool.

Closes #7293 (to reproduce issue easily, modify Util.func_240979_a_ and change the values in the MathHelper.clamp call). I've verified that 3 and below for "max" cause the problem. (I didn't test a whole range of values, just sufficient to reproduce problem and verify fix). Note that setting it to zero (and using the "direct executor" that's inaccessible in normal operation) works as well with this fix.

Signed-off-by: cpw <cpw+github@weeksfamily.ca>

tterrag1098 committed:
Add validation via resources for tag providers (#7271)

BREAKING CHANGE: Moved ExistingFileHelper to common package
Remove various workarounds from forge tag providers

tterrag1098 committed:
Fix broken ExistingFileHelper import

Add licenses to some new files

LexManos committed:
Dirty casting hacks to fix eclipse compiler inference issue. I'm sorry cpw.

sciwhiz12 committed:
Fix crash caused by previous commit (#7298)

cpw committed:
Fix compiler issue in eclipse properly

Signed-off-by: cpw <cpw+github@weeksfamily.ca>

cpw committed:
Use new enumeration mechanism in ModLauncher, to allow getResources to work. This enables serviceloader-in-mods, and other stuff that might need to visit multiple mod jars. Also, tweaked the visitor code slightly, may result in a trivial performance change. Closing #7302 as it's not really relevant any more.

Signed-off-by: cpw <cpw+github@weeksfamily.ca>

cpw committed:
Park the polling thread for a bit, if we're not the one driving the task list forwards. This allows for the actual driver to work on low cpu count machines.

Signed-off-by: cpw <cpw+github@weeksfamily.ca>

cpw committed:
Fix some potential issues in crash dumping during mod loading

Signed-off-by: cpw <cpw+github@weeksfamily.ca>

cpw committed:
Fix ExplodedDirectoryLocator visiting non-existent paths. Closes #7203

Also, bump modlauncher to 7.0.1 to fix resources.

Signed-off-by: cpw <cpw+github@weeksfamily.ca>

cpw committed:
Tweak crash report dump to visit all the causes up to the top.

Signed-off-by: cpw <cpw+github@weeksfamily.ca>

ichttt committed:
Fix crash when dumping crash report for an exception that has a null cause (#7308)

Cyborgmas committed:
Add dataprovider for Global Loot Modifiers (#6960)

ichttt committed:
Reimplement FarmlandWaterManager (#7213)

ChampionAsh5357 committed:
Re-implement DrawHighlightEvent$HighlightEntity Firing (#7233)

sciwhiz12 committed:
Remove dead ForgeHooks.canToolHarvestBlock function. (#7262)

sciwhiz12 committed:
Bump resource pack version to 6 (#7299)

EdwinMindcraft committed:
[1.16.x] Fixes for Bamboo, Enchantments and Conduits, small adjustments to bring some patches closer to the vanilla code. (#7239)

nrbeech committed:
Fixing logic in getCurrentRailPosition to correctly identify the rail… (#7276)

Tschipp committed:
Add partialTicks to RenderNameplateEvent (#7277)

Cyborgmas committed:
Prevent duplicate wrapper tags from crashing (#7283)

mrp-v2 committed:
Fix custom teleporters NPE (#7296)

Cyborgmas committed:
Fixed crash with zip paths (#7300)

ChampionAsh5357 committed:
New hook for better custom scaffolding movement (#7261)

cpw committed:
More crash report tweaks. Put a button to open the generated crash report on the error screen, tweak formatting of crash report, and add the enhanced stack trace data (transformers et al)

Signed-off-by: cpw <cpw+github@weeksfamily.ca>

pupnewfster committed:
Add support for custom tag types (#7289)

diesieben07 committed:
Fix Minecart speed cap on rail being initialized to 0 (#7315)

cpw committed:
More crash reporting tweaks. Don't crash when trying to show warnings. Also, put the exception name in the error screen on the second line!

Signed-off-by: cpw <cpw+github@weeksfamily.ca>

malte0811 committed:
Fix exception when getting rendering box for tile entities with no collision boxes. (#7301)

RedstoneDubstep committed:
Fix tile entities being replaced when not needed. (#7318)

LexManos committed:
1.16.3 Update
Also included a bunch of warning cleanups.

LexManos committed:
Bump MCP version for lambda issue.

LexManos committed:
New BiomeLoadingEvent that allows modders to edit biomes as they are being deserialized.

Yunus1903 committed:
Reimplemented drawHoveringText (#7268)

gigaherz committed:
Fix inconsistencies in how the values from the model are passed into the baked model.
This puts them in line with vanilla, as intended.

Thelonedevil committed:
Fix json biomes not setting registry name correctly for the BiomeLoadingEvent. (#7329)

Cyborgmas committed:
Add MatrixStack-aware alternatives to the methods in GuiUtils (#7127)

cpw committed:
Use mixin 0.8.1. Should fix problems with mixin not working properly with latest modlauncher.

Signed-off-by: cpw <cpw+github@weeksfamily.ca>

alcatrazEscapee committed:
Make Biome.Climate and BiomeAmbiance fields accessible (#7336)

alcatrazEscapee committed:
[1.16] Reimplement ICloudRenderHandler, ISkyRenderHandler and IWeatherRenderHandler (#6994)

Nightenom committed:
[1.16] Fix possible crash when using rendering regionCache (#7207)

sciwhiz12 committed:
Fix parse results of CommandEvent being disregarded (#7303)

Darkhax committed:
Fixed container item being consumed in brewing stand. #7307 (#7314)

ichttt committed:
Add a better license error screen (#7350)

Yunus1903 committed:
Updated build badge versions to 1.16.3 in readme (#7325)

diesieben07 committed:
Fix forge light pipeline applying block offsets twice (#7323)

diesieben07 committed:
Fix ToggleableKeyBinding differences to vanilla. (#7338)

pupnewfster committed:
Fix a few minor issues with custom tag types and fix OptionalNamedTags (#7316)

LexManos committed:
Revive BiomeManager and BiomeLayer hooks.

alcatrazEscapee committed:
Accessors for inspecting and removal of biome generation settings. Fixed carvers still being immutable. (#7339)

ichttt committed:
Fix grass disappearing when alwaysSetupTerrainOffThread is true (#7291)

LexManos committed:
Prep new RB.

cech12 committed:
Support effective materials of vanilla AxeItem (#7351)

DaemonUmbra committed:
Concluded rather messy rebase and merge

DaemonUmbra commented:
Well that merge was an unmitigated disaster
